analysis/src/zeroes.rs

```diff
@@ -0,0 +1,52 @@
+// See analysis/plot_zeroes
+
+extern crate core;
+extern crate rand;
+extern crate xorf;
+
+use core::convert::TryFrom;
+use rand::Rng;
+use xorf::*;
+
+const SIZE: usize = 7_000_000;
+
+macro_rules! print_zeroes {
+    ($filter:ident) => {{
+        let mut keys: Vec<u64> = Vec::with_capacity(SIZE);
+
+        for _ in 0..SIZE {
+            let key: u64 = rand::thread_rng().gen();
+            keys.push(key);
+        }
+        let filter = $filter::try_from(keys).unwrap();
+        let fp = filter.fingerprints();
+        let window_size = 2000;
+        let mut zeroes: usize = 0;
+        for (i, t) in fp.iter().enumerate() {
+            if *t == 0 {
+                zeroes += 1;
+            }
+            if i > window_size && fp[i - window_size] == 0 {
+                zeroes -= 1;
+            }
+            if i > window_size && i % 333 == 0 {
+                println!("{}", zeroes);
+            }
+        }
+    }};
+}
+
+fn main() {
+    let filter_str = std::env::args()
+        .collect::<Vec<_>>()
```

Comment:
this is a weird typing. Are you saying its a vec of any type?

Reply:
No, I'm asking rustc to infer the type for me